src/lane.cpp

```diff
@@ -1,28 +1,68 @@
 // Header
 #include "lane.hpp"
 
-// stlib
-#include <vector>
-#include <algorithm>
-
-// Releases all graphics resources
-bool Lane::init(float world_scale)
+Lane::Lane(float world_scale, direction dir)
 {
-	m_cars = new Car[MaxCarsPerLane];
-	return true;
+	m_world_scale = world_scale;
+	m_dir = dir;
+	m_time_remaining = m_max_time_per_car;
 }
 
-void Lane::destroy()
+Lane::~Lane()
 {
-	delete m_cars;
+	m_cars.clear();
 }
 
 float Lane::get_time_remaining() const
 {
     return m_time_remaining;
 }
 
-Car* Lane::get_cars() const
+std::vector<Car> Lane::get_cars() const
 {
     return m_cars;
 }
+
+bool Lane::update(float ms)
+{
+	m_time_remaining -= ms;
+	if (m_time_remaining <= 0)
+	{
+		this->turn_car();
+		m_time_remaining = m_max_time_per_car;
+	}
+	return true;
+}
+
+void Lane::add_car()

```

Comment:
LaneManager will determine what car will make the game harder, and will tell that lane to add that specific car. Will probably need to update Lane::add_car() to take in a parameter of a carType?

Reply:
hmm you're right. What will carType be? Are we going to create an enum that lists a bunch of carTypes (that correspond to different classes) or is there some way to pass a type (but not an object) as a parameter in C++?

I feel like the enum would be better because then we can fine tune it so different values correspond to cars with different initialization parameters (sprite, size, etc.) rather than just different classes

Reply:
Yeah I think we'll need to have carType as an enum. When lane calls Car::init(carType), it will then need to instantiate a car with the correct parameters and mesh/texture. That'll be for another PR though.